core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -39,7 +39,7 @@ private V2Metadata() {}
           ManifestFile.MANIFEST_CONTENT.asRequired(),
           ManifestFile.SEQUENCE_NUMBER.asRequired(),
           ManifestFile.MIN_SEQUENCE_NUMBER.asRequired(),
-          ManifestFile.SNAPSHOT_ID.asRequired(),
+          ManifestFile.SNAPSHOT_ID,
```

Comment:
I believe that this is actually required because if the manifest isn't written with a snapshot ID, the manifest list's snapshot ID is inherited. And that _must_ be present.

Reply:
Yes this is the case, but since the `ManifestFile.SNAPSHOT_ID` is already required:

https://github.com/apache/iceberg/pull/8673/files#diff-063af8055f56892097a498f7de319845da2e0d328f8241a5fc1b88a0822ac08fL51-L52

There is no need to mark it as required again. With the change in this PR, it is required for both V1 and V2, we don't need to mark it required anymore when constructing the individual schemas for V1 and V2.

Reply:
Keep in mind that the variable name is very confusing, it actually refers to the `ADDED_SNAPSHOT_ID`